cli/templatepush.go

```diff
@@ -275,13 +275,13 @@ func (pf *templateUploadFlags) setWorkdir(wd string) {
 	}
 }
 
-func (pf *templateUploadFlags) stdin() bool {
-	return pf.directory == "-"
+func (pf *templateUploadFlags) stdin(inv *serpent.Invocation) bool {
+	return pf.directory == "-" || (!isTTYIn(inv) && pf.directory == "")
```

Comment:
When this condition is tripped it should probably log something along the lines of "Receiving template tar from stdin"